examples/mixture/plot_gmm_selection.py

```diff
@@ -22,7 +22,7 @@
 import matplotlib.pyplot as plt
```

Comment:
I think running all the models initialized with different agglomerative parameters (linkage, affinity) would cause it to run longer. Actually, there were 264 models fitted when running GaussianMixtureIC. And the results are indeed different here: many of the bic values are smaller. Here are the raw bic values.

on main
```
array([9451.85111673, 6757.39900009, 6788.06648909, 6028.00869765,
       5762.83542461, 5788.46680114, 8009.34258683, 6226.95905566,
       6041.65816023, 5683.87395387, 5563.77301084, 5279.82469731,
       9000.25372828, 5991.40332284, 6024.86246735, 5342.66326919,
       5374.12672816, 5186.64701626, 8009.34258683, 4560.77070969,
       4604.71454071, 4639.43580943, 4669.02214548, 4715.84537213])
```

on this branch
```
array([9451.85111673, 6757.39900009, 6425.87046435, 6028.25984227, 
       5762.51264199, 5789.56990116, 8009.34258683, 6226.95905566,
       5869.79866258, 5590.88971744, 5408.2608514, 5247.55509724,
       9000.25372828, 5991.40332284, 5683.50768211, 5341.94173417,
       5152.39220985, 5103.63717724, 8009.34258683, 4560.77070969,
       4571.37424516, 4621.94460561, 4656.81584998, 4694.44150557])
```
But please let me know if you think we should investigate further for this example.

Reply:
> I think running all the models initialized with different agglomerative parameters (linkage, affinity) would cause it to run longer. Actually, there were 264 models fitted when running GaussianMixtureIC

OK it makes sense for the duration then.

Regarding BICs, they are still relatively close even-though much more models are fitted in this PR case. I am not sure we want to introduce `GaussianMixtureIC` it if there's a substantial longer running time compared to the simple double-loops snippet just above. Would it be possible to initialize models in a simpler fashion?

I would prefer having someone else's opinion on this contribution.


(Also, have I switched the two branches' results just above? 🤔)

Reply:
cc @tliu68

Reply:
Sorry for the late reply! I believe this has been fixed by using no agglomeration as default (as in [this commit](https://github.com/scikit-learn/scikit-learn/pull/19562/commits/1da945df4587852c1c91aee7116e96b9e12f88ee))

Reply:
Indeed, I've checked again and this is fixed.